src/contexts/applicationContext.tsx

```diff
@@ -104,6 +107,20 @@ function ApplicationProvider({ children }: ApplicationProviderProps) {
     [setSelection]
   )
 
+  const changeSearchValue = useCallback(
+    (term: string) => {
+      setSearchValue(term)
+    },
+    [setSearchValue]
+  )
+
+  const changeMacroOutputEnabled = useCallback(
+    (value: boolean) => {
+      setIsMacroOutputEnabled(!value)
+    },
+    [setIsMacroOutputEnabled]
+  )
+
```

Comment:
I'm not loving the fact that this is in this "ApplicationContext". It's primarily just for searching the macro collections right? It should be located closer to that if so and not using context

Reply:
The idea was that its tracked in the application context since you can only have one ongoing search at a time, i could try finding a closer spot to it, tho, it is useful to share the isSearching and other data (or have it accessible from multiple places) i think. Will deffo be for the macro output for example when testing macros